lib/WUI/netdev.c

```diff
@@ -222,22 +222,34 @@ netdev_status_t netdev_get_status(uint32_t netdev_id) {
     }
 }
 
-netdev_ip_obtained_t netdev_get_ip_obtained_type() {
-    return IS_LAN_STATIC(ETH_CONFIG().lan.flag) ? NETDEV_STATIC : NETDEV_DHCP;
+netdev_ip_obtained_t netdev_get_ip_obtained_type(uint32_t netdev_id) {
+    return IS_LAN_STATIC(wui_netdev_config[netdev_id].lan.flag) ? NETDEV_STATIC : NETDEV_DHCP;
 }
 
 uint32_t netdev_set_dhcp(uint32_t netdev_id) {
+    ETH_config_t *pConfig = NULL;
+    err_t res = ERR_OK;
+
     if (netdev_id == NETDEV_ETH_ID) {
-        err_t res;
         res = netifapi_dhcp_start(&eth0);
-        CHANGE_FLAG_TO_DHCP(wui_netdev_config[netdev_id].lan.flag);
-        wui_netdev_config[netdev_id].var_mask = ETHVAR_MSK(ETHVAR_LAN_FLAGS);
-        save_eth_params(&wui_netdev_config[netdev_id]);
-        wui_netdev_config[netdev_id].var_mask = 0;
-        return res;
+        pConfig = &wui_netdev_config[netdev_id];
+    } else if (netdev_id == NETDEV_ESP_ID) {
+        esp_sta_quit(NULL, NULL, 1);
+        esp_sta_join(ap.ssid, ap.pass, NULL, 0, NULL, NULL, 0);
+        pConfig = &wui_netdev_config[netdev_id];
+        res = ERR_OK;
+    }
+
+    if (pConfig) {
+        CHANGE_FLAG_TO_DHCP(pConfig->lan.flag);
+        pConfig->var_mask = ETHVAR_MSK(ETHVAR_LAN_FLAGS);
+        save_eth_params(pConfig);
```

Comment:
:question: Will the EEPROM have the _same_ slot for both ethernet and wifi? Because right now no matter which one we save, it goes to the same place and doesn't allow us to have the future „both“ mode.